### DISCUSSION TITLE
How to actually utilise source maps?

### DISCUSSION THREAD

**Discussion by MC-DeltaT** [parcel-bundler/parcel#6477] [Q&A] [answered]
I'm new to Parcel and web bundling in general, and have come across a seemingly very basic issue regarding how to use the source maps produced by Parcel.

I am trying to create a very simple static site, with a Node JS & Express server.

My project structure looks like this:
<pre>
- project_root
    - web_content (all static web content)
        - scripts
            - foo.js (JS module used by index.js)
        - index.html (web entrypoint)
        - index.js (JS for index.html)
    - other stuff (back end source code, node_modules, package.json, etc.)
</pre>

Before considering web bundling, I would just serve the entire `web_content` directory as a static site with Express, which would be perfectly fine.

Now bundling with Parcel, I use the following command:
`npx parcel build web_content/index.html --dist-dir web_content_build --public-url /web`
Which would create a directory `web_content_build` in the project root, containing the bundled data. I figure I can serve this `web_content_build` as a static site with Express, just like before. (The `--public-url /web` is because I wish the static site to be hosted at `/web`.)

This works mostly OK, except that the source maps generated by Parcel seem to point back to the original `web_content` directory, which obviously is not served by the server.
For example, the source map for `index.js` references the path `./web_content/index.js`. That path doesn't exist within the generated `web_content_build` directory, so it doesn't get served and hence the source map doesn't work.

This seems like such a trivial issue, but I can't find anyone else encountering a similar thing. Am I setting up the project structure "incorrectly"? What is the intended way to get source maps to work?

Solutions I have considered:
 - Serving the entire project directory with Express, so everything is accessible. This seems unnecessary and just plain dumb.
 - Making Parcel generate `web_content_build` within `web_content`, and then serving `web_content` with Express. I don't like this because then the URLs are messed up. I want to maintain a good URL structure.
 - Copying everything from `web_content` into `web_content_build` first (either manually or with a Parcel plugin), then running the Parcel build to add in the bundled stuff. I guess this works, but then why doesn't Parcel do this anyway so the source maps work by default?

**Comment by mischnic** [comment, marked as answer]
You can do 
```js
{
 "targets": {
  "default": {
   "sourceMap": {
     "inlineSources": true
   }
  }
 }
}
```

then the sourcemap files contain the actual source contents as a string.

Or set `sourceRoot`  so that the relative url from the sourcemap file to your sourcefiles is correct: 
https://v2.parceljs.org/configuration/package-json/#sourcemap

**Reply by MC-DeltaT** [reply]
Ah, thank you, I did not realise this was configurable.
I could only find the docs for Parcel V1 - the V1 vs. V2 thing is quite confusing unless you know exactly what you are looking for.
